src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -63,6 +63,10 @@ public void channelRead0(ChannelHandlerContext ctx, Batch batch) throws Exceptio
                 }
             }
         }finally{
+            ctx.channel().attr(ConnectionHandler.CHANNEL_PENDING_COUNTER).get().getAndDecrement();
```

Comment:
Spoke in person, and for very large batches, it can read from org.logstash.beats.ConnectionHandler#channelRead multiple times, and due to the decoder converting the read bytes to a org.logstash.beats.Batch it will only call the org.logstash.beats.BeatsHandler#channelRead0 once...which throws off the counts. 

Since in this case, we only really care about on/off for the keep alive, we should be able to adjust this code to use booleans instead of counting to the same effect. 

Also need to turn off keep alive if a pipeline exception occurs.